jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -37,6 +37,7 @@ JERRY_STATIC_ASSERT (sizeof (jmem_cpointer_t) <= sizeof (ecma_value_t),
 
 #ifdef ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY
 
+/* cppcheck-suppress zerodiv */
```

Comment:
> To be honest, I'm not really fond of inline cppcheck-specific warning suppression comments in the C code. They either signal something bad that should be fixed, or they should be listed in the suppression list file if they are false positives. IMHO.

I think adding a suppression anywhere should always mean false positive warning. If it signals a real bug, it should be fixed instead of suppressing. In exceptional cases I can imagine suppressing a warning until fixing, but in this case I would add at least a FIXME comment and file a bug report.

I'd rather add inline suppress comments, because they aren't sensitive to linenumber changes. I'm afraid if we added line specific suppressions to cppcheck/suppression-list file, we should update line numbers in it too frequently.

Reply:
Of course I can move the remaining 4 inline suppression to the cppcheck/suppression-list, if do you still prefer that solution. Or maybe we should add comment for the inline suppression to clarify that they are false warnings, for example:
/* cppcheck-suppress <warning-type> ; false cppcheck warning */